ff/src/fields/models/cubic_extension.rs

```diff
@@ -372,8 +389,11 @@ impl<P: CubicExtParameters> FromBytes for CubicExtField<P> {
 impl<P: CubicExtParameters> Neg for CubicExtField<P> {
     type Output = Self;
     #[inline]
-    fn neg(self) -> Self {
-        Self::new(self.c0.neg(), self.c1.neg(), self.c2.neg())
+    fn neg(mut self) -> Self {
+        self.c0 = -self.c0;
```

Comment:
should this instead be calling neg on the base field?

Reply:
`neg` is just the method name corresponding to `-`.